bindings/Sofa/tests/Core/Events.py

```diff
@@ -0,0 +1,57 @@
+# coding: utf8
+
+import unittest
+import Sofa
+import Sofa.Core
+import Sofa.Simulation
+import SofaRuntime
+
+
+class MyController(Sofa.Core.Controller):
+    """This is my custom controller
+        when init is called from Sofa this should call the python init function
+    """        
+
+    def __init__(self, *args, **kwargs):
+        ## These are needed (and the normal way to override from a python class)
+        Sofa.Core.Controller.__init__(self, *args, **kwargs)
+        self.inited = 0
+        self.build_iterations = 0
+        self.solve_iterations = 0
+
+    def __del__(self):
+            pass
+    
+    def init(self):
+            self.inited += 1
+
+    def onEvent(self, kwargs):
+            pass
+
+    def onBuildConstraintSystemEndEvent(self, kwargs):
+            self.build_iterations+=1
```

Comment:
I don't understand how this function can be called if there is no constraint solver in the scene